src/components/carousel/nz-carousel.component.ts

```diff
@@ -83,14 +96,18 @@ export class NzCarouselComponent implements AfterViewInit, OnDestroy {
       if (this.nzAutoPlay) {
         this.createInterval();
       }
-      this._renderer.setStyle(this.slickList.nativeElement, 'height', `${this.hostElement.nativeElement.offsetHeight}px`);
+
       if (this.nzVertical) {
+        this._renderer.setStyle(this.slickList.nativeElement, 'height', `${this.hostElement.nativeElement.offsetHeight}px`);
```

Comment:
@vthinkxie if not, the height is 0,  `nz-demo-carousel-vertical` demo can't display normally

you should test it by modify the code:


*NzDemoCarouselVerticalComponent*

```
export class NzDemoCarouselVerticalComponent implements OnInit {
 array = [  ]; // try dynamic change the array
  
    constructor() {
    }
  
    ngOnInit() {
  setTimeout(()=>{
      this.array = [ 1, 2, 3, 4 ];
    },500)
    }
  }

```